servers/rendering/shader_language.cpp

```diff
@@ -5549,9 +5549,22 @@ ShaderLanguage::Node *ShaderLanguage::_parse_expression(BlockNode *p_block, cons
 									}
 									if (is_sampler_type(call_function->arguments[i].type)) {
 										//let's see where our argument comes from
-										ERR_CONTINUE(n->type != Node::NODE_TYPE_VARIABLE); //bug? this should always be a variable
-										VariableNode *vn = static_cast<VariableNode *>(n);
-										StringName varname = vn->name;
+
+										StringName varname;
+										if (n->type == Node::NODE_TYPE_VARIABLE) {
+											VariableNode *vn = static_cast<VariableNode *>(n);
+											varname = vn->name;
+										} else if (n->type == Node::NODE_TYPE_ARRAY) {
+											ArrayNode *an = static_cast<ArrayNode *>(n);
+											if (is_sampler_type(an->datatype_cache)) {
+												varname = an->name;
+											} else {
+												ERR_CONTINUE_MSG(true, "The argument passed in is not a sampler.");
+											}
+										} else {
+											ERR_CONTINUE_MSG(true, "The argument passed in is not a sampler.");
+										}
+
```

Comment:
```suggestion
										StringName varname;
										if (n->type == Node::NODE_TYPE_VARIABLE) {
											VariableNode *vn = static_cast<VariableNode *>(n);
											varname = vn->name;
										} else if (n->type == Node::NODE_TYPE_ARRAY) {
											ArrayNode *an = static_cast<ArrayNode *>(n);
											varname = an->name;
										}
```

Type checking occurs before that code, so there is no reason to check it detailed and emit errors here.

Reply:
Thanks!